docs(changelog): add 8.4.0 release notes (#5027)
canonical/snapcraft: docs/reference/changelog.rst

  For a complete list of commits, check out the `X.Y.Z`_ release on GitHub.

8.4.0 (2024-Sep-10)
-------------------

.. note::

   8.4.0 includes changes from the :ref:`7.5.6<7.5.6_changelog>` release.

Core
====

> * Fixes a bug where Snapcraft would fail to inject itself into the build

Reviewer: > Fixes

These notes use a different tense than past releases. They should be consistent. Did we change the underlying guidelines for commit messages?

Author: We didn't.  I wanted to see what a changelog looks like in the indicative mood instead of imperative.  

I like inidicative better but I switched it back for consistency.

Reviewer: I don't see a change?

Author: Whoops, forgot to push the commit.  Should be changed now

  environment when not running as a snap (`canonical/charmcraft#406`_). If an
  app isn’t running from snap, the installed app will now install the snap in
  the build environment using the channel in the ``CRAFT_SNAP_CHANNEL``
  environment variable, defaulting to ``latest/stable`` if none is set.

> * Fixes a regression where icons would not be configured and installed for

Reviewer: ```suggestion
* Fixes a regression where icons wouldn't be configured and installed for
```

  snaps with no ``apps`` defined in their ``snapcraft.yaml``.

Bases
#####

core24
""""""

* Raises an error if the build plan is empty and no snaps will be built
  (`canonical/craft-application#225`_).

* Fixes a regression where ``https_proxy``, ``https_proxy``, and ``no_proxy``
  were not forwarded into the build environment.

Plugins
#######

* Fixes a bug where ``snapcraft list-plugins --base core20`` would fail in a
  ``core24`` project directory (`#5008`_).

Components
##########

* Allows numbers and hyphens in component names (`LP#2069783`_).

> * Fixes a bug where ``stage-packages`` could not be used when components are

Reviewer: ```suggestion
* Fixes a bug where ``stage-packages`` can't be used when components are
```

  defined (`canonical/craft-parts#804`_).

Command line
============

* Improves error messages when parsing a ``snapcraft.yaml`` file (`#4941`_).

* Improves error messages when using an `ESM base`_.

* Improves error messages for missing files (`canonical/craft-parts#802`_).

> * Improves error messages when a build fails due to matching multiple platforms
>   (`canonical/craft-application#382`_).

Reviewer: ```suggestion
* Improves error messages when a build fails because it matches multiple
  platforms (`canonical/craft-application#382`_).
```
